lib/Runtime/Executor/ThreadPoolExecutor.cpp

```diff
@@ -207,6 +194,8 @@ void ThreadPoolExecutor::handleDeviceManagerResult(
       }
     }
   }
+  // Return intermediateContext to executionState.
+  executionState->returnUniqueNodeContextPtr(node, std::move(ctx));
```

Comment:
I found this context check in and check out a bit awkward. Maybe if we use raw point in the lower level calls can improve the situation. You don't need to change this now.